lib/mongo/collection/view/change_stream.rb

```diff
@@ -176,15 +247,30 @@ def for_collection?
         end
 
         def cache_resume_token(doc)
+          # Always record both resume token and operation time,
+          # in case we get an older or newer server during rolling
+          # upgrades/downgrades
           unless @resume_token = (doc[:_id] && doc[:_id].dup)
-            raise Error::MissingResumeToken.new
+            raise Error::MissingResumeToken
           end
         end
 
         def create_cursor!
+          # clear the cache because we may get a newer or an older server
+          # (rolling upgrades)
+          @start_at_operation_time_supported = nil
+
           session = client.send(:get_session, @options)
           server = server_selector.select_server(cluster)
           result = send_initial_query(server, session)
+          if doc = result.replies.first && result.replies.first.documents.first
```

Comment:
Is this the command response, or one of the changes? operationTime comes from the outer command response document.

Reply:
This is the command response; just tried it out locally, which confirmed what I understood to be how we used `send_initial_query` methods in our various operations